### ISSUE TITLE
Bot attempting to snipe the same time over and over again

### ISSUE THREAD

**Issue by msieb1** [Alkaar/resy-booking-bot#103]
Hello, thanks for the work you have put into this repo! I verified that the bot worked on a restaurant with available reservations, for which I was successfully able to have the bot make a reservation.

However, I've encountered an issue for harder-to-get places: For example, I provided a list of times to book for "I Sodi" in Manhattan, but the bot seems to be stuck on a single time during the sniping process.

My config is:
```python
resDetails.res-time-types=[
    {reservation-time="20:00:00"},
    {reservation-time="20:30:00"},
    {reservation-time="21:00:00"}
    {reservation-time="21:30:00"}
    {reservation-time="22:00:00"},
]
```

It doesn't attempt the 20:00 or 20:30 - I suppose they were already gone. It does attempt the 21:00 though:
```
[INFO ] 2023-03-30 20:59:58.023-07:00 ResyBookingWorkflow:20 - Attempting to snipe reservation
[INFO ] 2023-03-30 21:00:00.240-07:00 ResyClient:197 - Config Id: rgs://resy/43/116476/2/2023-04-13/2023-04-13/21:00:00/2/Dining room
...
[INFO ] 2023-03-30 21:00:00.370-07:00 ResyClient:88 - Book Token: |OEpux4HCgi_8vx3||5Ofz6xQqU_Z7q12FNoFYbtILFXsoT|IV8HBLoaI60VN5J|OI6kfCqFA7NxS6PKKH0kvPjAywTFYQ_C1XDTsdXAusehEQBREMqZ9HxUTN3HnMMujn|Kz0XlXDFFxjML1tpxOxhAjc3cPaF1b3rNkR|RFYTve562RzqctmwNy2EhGIueoJvG70fis5ImdyzElIV_CMExJE8KCOw88SdJuW9SosLU2FHpH01zib7bcM0QeLNje1d3UzjJj0uDfsnOikNfDUpeqpXY_Cyu7MSFalvCormHwxCmCBEOxtkLDWBtIT2F57P4MoYzlQ5TIwFnG0YGR07UrPAWLmK4Lmuk8fZMhWmtJJ|S3S1jdd5Whk57SJrShboLxocjRDL2O7Ya0Y_i6xc0QL80Al1sDx1UT2IjMeeypW77Z|quWbaHjXGmanLFlzDuoOQ7R8RfCQqFyBIQdP9ZpOrTUd4Lsa41EO2ivJCG0jI5iEHT0CPdAdcrFRcIuPf2|rpOvW32aBccpmW6kORQeSvVqFpEWVRMXJVGVTxAvI3HDh4fTr4NNQB4DMe-5779493384e6998caa0d72af8df2505b970319079c278ccedfcda686
[INFO ] 2023-03-30 21:00:00.460-07:00 ResyClient:136 - Missed the shot!
[INFO ] 2023-03-30 21:00:00.460-07:00 ResyClient:137 - ┻━┻ ︵ \(°□°)/ ︵ ┻━┻
[INFO ] 2023-03-30 21:00:00.461-07:00 ResyClient:138 - Reservation no longer available
```
So far so good. But instead of going to the next time, it keeps trying the 21:00 slot again (with a different `Book Token`):
```
[INFO ] 2023-03-30 21:00:01.074-07:00 ResyBookingWorkflow:18 - Taking the shot...
[INFO ] 2023-03-30 21:00:01.074-07:00 ResyBookingWorkflow:19 - (҂‾ ▵‾)︻デ═一 (˚▽˚’!)/
[INFO ] 2023-03-30 21:00:01.075-07:00 ResyBookingWorkflow:20 - Attempting to snipe reservation
[INFO ] 2023-03-30 21:00:01.297-07:00 ResyClient:197 - Config Id: rgs://resy/43/116476/2/2023-04-13/2023-04-13/21:00:00/2/Dining room
[INFO ] 2023-03-30 21:00:01.418-07:00 ResyClient:79 - Payment Method Id: 14787425
[INFO ] 2023-03-30 21:00:01.419-07:00 ResyClient:88 - Book Token: J8GC|GsM3MSrpyjj_z_wCxF7vegnE3RoqUTPlmmOoxN5sS4vHCWVz9GC0HIb|beCgddcYL2oDXRwtAozR62erOlBxisDuV3SqoM4hVodLXPTgoBMkVgQnb9MEa3ORRElySBCcfU|DeQV9aLoQ4qPWioDF3Ak2201O6LB4sYYf5Jc55qpZpJK5tMycE4_cKWri2S2cbz9X3cwCU3tegGggdTWaV5ZAK8Gl9yuVtCRv3T51|2yXI1wb9ycMVIMXqfyx_Sx6igyjSBtFmoeGZm6QX5NmnElOwRtxOL8fIYfvYNobONCaj|3j6gSZsrbQAMe0IIDr3ftweiLLjnX49ajn6yRoAn173sdCHWcKXfsA8qFRUaZwCiWsVlAYgo4ryg8ObaMOtqEla9qjZm8OC8XOEDjuTNF1Wk9wSrS|WzOwedNcyGevpx0nARZAFK_b3S8h9wVyj07VqvCTB|6kfoDS9ejSLIy8_r9VGX2ymjWYAyDSdy4oZH_G8uFqoOXMWUYzztzn2Tl4XuySUZSRgJ0j4HK48ne4raAzIe6MrxnxRxtQUHUJ58tige0_LgxrtZdt-a8d9c1f9fc93fea574a230d6c757cd236976c423dfc2a0d2dacec0ef
[INFO ] 2023-03-30 21:00:01.500-07:00 ResyClient:136 - Missed the shot!
[INFO ] 2023-03-30 21:00:01.501-07:00 ResyClient:137 - ┻━┻ ︵ \(°□°)/ ︵ ┻━┻
[INFO ] 2023-03-30 21:00:01.501-07:00 ResyClient:138 - Reservation no longer available
```

And this keeps on going for another 10 times or so for the same 21:00 slot, however always with a different `Book Token`, before eventually shutting down. Is this intended behavior? Why is the bot not proceeding to the next time? Where in the code can I set a threshold on how often to try for the same time before proceeding to the next one? For all I know, it might've gotten the 21:30 reservation slot if it had simply moved on after missing the first 21:00 snipe attempt.

**Comment by suera23** [comment]
thanks Alkaar for this bot!!!  I've used it successfully in the past, but am now having the same/similar issue as msieb1 above.  I saw the discussion item re: the payment service migration so made a reservation at a place that requires a deposit.  Also tried with a new credit card - still same issue.   I'm trying to test this out on place that has reservations available, so I know I did not really 'miss the shot'.  Any thoughts on things to look at?  Thanks again!
